doc/contributing/submitting-changes.chapter.md

```diff
@@ -234,6 +234,21 @@ digraph {
 
 The `master` branch is the main development branch. It should only see non-breaking commits that do not cause mass rebuilds.
 
+What is a mass rebuild? It's an update that takes Hydra build infrastructure long time to process (a few days).
+
+Here are rules of thumb that should help you decide if a change is reasonable for `master` branch:
+
+- Small update that rebuilds under 100 dependent packages unlikely to break users' systems. Examples are leaf package updates.
+- Important update of heavyweight leaf package like `chromium` or `firefox`
+- Important update with rebuild count less than 1000 dependent packages. Batch close to 1000 should not contain heavyweight packages like `chromium` or `firefox`.
+- Kernel updates
+- Urgent security fixes
+
+Otherwise consider `staging` branch. Typical changes for `staging` are:
+
+- Any changes to `stdenv` (trigger full rebuld)
+- Any core language ecosystem changes like `python` or `ghc` update (usually trigger more than 1000 dependent packages rebuild)
```

Comment:
`ghc` updates are done independently on `haskell-updates`, so a bad example. Large scale python changes are often done using `python-unstable`. I think language infrastructure is a bad example in general, since they are quite specific.

Better examples would be core libraries I think (gmp, ...).

Reply:
`master`, but that's not really important here. Currently working on haskell documentation anyways and we'll just cover the haskell contributing quirks there, I think.

Reply:
> ghc updates are done independently on haskell-updates, so a bad example.

Maybe in this case but they should be mentioned so that new contributors have a basic guiding.

Reply:
I don't think it makes sense here — except if you want to overwhelm the reader. There are a lot of differences wrt workflow for certain subsystems of nixpkgs. If you want to make contributions to the haskell ecosystem, you'll want to read the (currently lacking) haskell contribution docs anyways. Besides, so far there have been zero occasions of someone PR-ing a GHC change on their first day…